torchmetrics/metric.py

```diff
@@ -221,6 +231,72 @@ def wrapped_func(*args, **kwargs):
 
         return wrapped_func
 
+    def sync(
+        self,
+        dist_sync_fn: Optional[Callable] = None,
+        process_group: Optional[Any] = None,
+        should_sync: bool = True,
+    ) -> Dict[str, Tensor]:
+        """
+        Sync function for manually controlling when metrics states should be synced across processes
+
+        Args:
+            dist_sync_fn: Function to be used to perform states synchronization
+            process_group:
+                Specify the process group on which synchronization is called.
+                default: None (which selects the entire world)
+            should_sync: Whether to apply to state synchronization.
+
+        Returns:
+            cache: A dictionarry containing the local metric states. The cache will be empty if sync didn't happen.
+        """
+        is_distributed = torch.distributed.is_available() and torch.distributed.is_initialized()
```

Comment:
Can you move this to a function? Will need to override it for TPUs or other accelerators

Reply:
Yes, good catch !